### PR TITLE
extending WebDriverException

### LINKED ISSUE #11712: [🐛 Bug]: org.openqa.selenium.InvalidSelectorException is not thrown by fluent wait even when it's not ignored
### What happened?

https://seleniumhq.slack.com/archives/C0ABCS03F/p1677391208456589

I'm using an invalid element locator (for testing purposes) and I'm trying to identify that element within a fluent wait
The fluent wait is ignoring some exceptions (not including org.openqa.selenium.InvalidSelectorException)
Expected Behavior: the invalid selector exception should be thrown immediately
Actual Behavior: it gets caught and rethrown recursively until a timeout occurs, and then a timeout is thrown with the cause InvalidSelectorException
Here's a gist with the fluentWait block, the expected exceptions block, and a screenshot from the throwable object.
https://gist.github.com/MohabMohie/6c4d16bfca0ad7b15c3e274b631ecdf9
Is this a bug or am I doing something wrong?

### How can we reproduce the issue?

```shell
https://gist.github.com/MohabMohie/6c4d16bfca0ad7b15c3e274b631ecdf9
```

### Relevant log output

```shell
N/A
```

### Operating System

Windows 11

### Selenium version

Java 4.8.1

### What are the browser(s) and version(s) where you see this issue?

N/A

### What are the browser driver(s) and version(s) where you see this issue?

N/A

### Are you using Selenium Grid?

N/A

### PR DESCRIPTION
**Pull request by MohabMohie**
- fixes #11712
- as per https://www.selenium.dev/meetings/2022/tlc-08-17/#proposals

**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] [I have updated the documentation accordingly.](https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1325)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: java/test/org/openqa/selenium/InvalidSelectorTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.openqa.selenium.testing.JupiterTestBase;
+
+class InvalidSelectorTest extends JupiterTestBase {
+
+  @Test
+  void testOldPage() {
+    driver.get(pages.simpleTestPage);
+    assertThatExceptionOfType(InvalidSelectorException.class).isThrownBy(driver.findElement(By.xpath("invalid-xpath"))::click);
+  }
+}
```
Comment (diemol) [inline java/test/org/openqa/selenium/InvalidSelectorTest.java]: This test is not needed. You need to check the `ElementFindingTest` class, and fix the failing tests, please.

**Review by diemol — CHANGES_REQUESTED** [review]

File: java/test/org/openqa/selenium/InvalidSelectorTest.java (hunk 1 above)
Comment (MohabMohie) [reply java/test/org/openqa/selenium/InvalidSelectorTest.java]: @diemol done, and reran all browser tests on 'chrome' and 'firefox' to confirm.
Thanks a lot for your support!
image

Comment (diemol) [reply java/test/org/openqa/selenium/InvalidSelectorTest.java]: Great!

**Review by diemol — APPROVED** [review]
Thank you, @MohabMohie!

**Pull request merged**